pytorch/training/docker/1.6.0/py3/Dockerfile.gpu

```diff
@@ -0,0 +1,184 @@
+
+WORKDIR /root
+
+RUN /opt/conda/bin/conda config --set ssl_verify False \
+ && pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org \
+ && ln -s /opt/conda/bin/pip /usr/local/bin/pip3
+
+# Uninstall torch and torchvision before installing the custom versions from an S3 bucket
+RUN pip install \
+    --no-cache-dir smdebug==${SMDEBUG_VERSION} \
+    "sagemaker<2" \
+    sagemaker-experiments==0.* \
+    --no-cache-dir "sagemaker-pytorch-training<3" \
+    --no-cache-dir fastai==1.0.59 \
+    "awscli<2" \
+    psutil \
+    Pillow \
+    scipy \
+ && pip install --no-cache-dir -U ${PT_TRAINING_URL} \
+ && pip uninstall -y torchvision \
+ && pip install --no-deps --no-cache-dir -U \
+     https://torchvision-build.s3.amazonaws.com/1.6.0/gpu/torchvision-0.7.0-cp36-cp36m-linux_x86_64.whl
+
+# Install Horovod
+RUN pip uninstall -y horovod \
+ && ldconfig /usr/local/cuda-10.1/targets/x86_64-linux/lib/stubs \
+ && HOROVOD_GPU_ALLREDUCE=NCCL HOROVOD_CUDA_HOME=/usr/local/cuda-10.1 HOROVOD_WITH_PYTORCH=1 pip install --no-cache-dir horovod==${HOROVOD_VERSION} \
+ && ldconfig
+
+# Install Nvidia Apex
+RUN git clone https://github.com/NVIDIA/apex.git \
+ && cd apex \
+ && pip install -v --no-cache-dir --global-option="--cpp_ext" --global-option="--cuda_ext" ./
+
+# Configure Open MPI and configure NCCL parameters
+RUN mv /home/.openmpi/bin/mpirun /home/.openmpi/bin/mpirun.real \
+ && echo '#!/bin/bash' > /home/.openmpi/bin/mpirun \
+ && echo 'mpirun.real --allow-run-as-root "$@"' >> /home/.openmpi/bin/mpirun \
+ && chmod a+x /home/.openmpi/bin/mpirun \
+ && echo "hwloc_base_binding_policy = none" >> /home/.openmpi/etc/openmpi-mca-params.conf \
+ && echo "rmaps_base_mapping_policy = slot" >> /home/.openmpi/etc/openmpi-mca-params.conf \
+ && echo "btl_tcp_if_exclude = lo,docker0" >> /home/.openmpi/etc/openmpi-mca-params.conf \
+ && echo NCCL_DEBUG=INFO >> /etc/nccl.conf \
+ && echo NCCL_SOCKET_IFNAME=^docker0 >> /etc/nccl.conf
+
+# Install OpenSSH for MPI to communicate between containers, allow OpenSSH to talk to containers without asking for confirmation
+RUN apt-get update \
+ && apt-get install -y  --allow-downgrades --allow-change-held-packages --no-install-recommends \
+ && apt-get install -y --no-install-recommends openssh-client openssh-server \
+ && mkdir -p /var/run/sshd \
+ && cat /etc/ssh/ssh_config | grep -v StrictHostKeyChecking > /etc/ssh/ssh_config.new \
+ && echo "    StrictHostKeyChecking no" >> /etc/ssh/ssh_config.new \
+ && mv /etc/ssh/ssh_config.new /etc/ssh/ssh_config \
+ && rm -rf /var/lib/apt/lists/*
+
+WORKDIR /
+
+ADD https://raw.githubusercontent.com/aws/aws-deep-learning-containers-utils/master/deep_learning_container.py /usr/local/bin/deep_learning_container.py
+
+RUN chmod +x /usr/local/bin/start_with_right_hostname.sh \
+ && chmod +x /usr/local/bin/deep_learning_container.py \
+ && curl -o /license.txt https://aws-dlc-licenses.s3.amazonaws.com/pytorch-1.6.0/license.txt
```

Comment:
Please add another revision to this PR changing the `curl` command here to a `wget`.

`curl -o <filename> <url>` doesn't return a non-zero error code when it fails to access the url.

`wget -O <filename> <uri>` returns a non-zero error code when it fails to access the url, telling us that the content at the url is either inaccessible or faulty.

Reply:
If you know of a way to make `curl` throw an error instead of failing silently, that would be a good change as well, instead of changing `wget` to `curl`.

Reply:
Addressed in next update.